tests/common/common_by_value.h

```diff
@@ -0,0 +1,242 @@
+/*******************************************************************************
+//
+//  SYCL 1.2.1 Conformance Test Suite
+//
+//  Copyright:	(c) 2017 by Codeplay Software LTD. All Rights Reserved.
+//
+// Provides verification for common by-value semantics
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_COMMON_COMMON_BY_VALUE_H
+#define __SYCLCTS_TESTS_COMMON_COMMON_BY_VALUE_H
+
+#include "common.h"
+
+#include <string>
+#include <array>
+#include <type_traits>
+
+namespace {
+
+/**
+ * @brief Check equality-comparable operations on the host side
+ */
+template <typename T>
+void check_equality_comparable_generic(sycl_cts::util::logger& log, const T& a,
+                                       std::string test_name) {
+  /** check for reflexivity
+   */
+  if (!(a == a)) {
+    FAIL(log, (test_name +
+               " is not equality-comparable (operator== reflexivity failed)")
+                  .c_str());
```

Comment:
This `FAIL` should be updated to take also `std::string` some day...

Reply:
Thank you for noticing. I've made changes by 9d88b50c4adc1c9351e30362077ded9caabc36e5. Also please take a look on fail_proxy changes by 4d664edf45fcf40ea9ae9492f1764f4e94db7b84